File: longest-consecutive-sequence/longestConsecutiveSequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        nums = set(nums)
        best = 0
        for i in nums:
            if i - 1 not in nums:
                j = i + 1
                while j in nums:
                    j += 1
                best = max(best, j - i)
        return best

File: longest-consecutive-sequence/longestConsecutiveSequence.java
public class Solution {
    public int longestConsecutive(int[] nums) {
        Map<Integer, Integer> map = new HashMap<>();
        int best = 0;
        for(int i : nums) {
            if(map.containsKey(i)) {
                continue;
            }
            int left = map.containsKey(i - 1) ? map.get(i - 1) : 0;
            int right = map.containsKey(i + 1) ? map.get(i + 1) : 0;
            int sum = left + right + 1;
            map.put(i, sum);
            best = Math.max(sum, best);
            map.put(i - left, sum);
            map.put(i + right, sum);
        }
        return best;
    }
}